### PR TITLE
Bugfix for completion for malformed input

### PR DESCRIPTION
**Pull request by insafuhrmann**
Bugfix for completion break due to malformed/incomplete input. Concerns #778.

### CODE REVIEW

File: packages/langium/src/utils/cst-util.ts [hunk 1]
```diff
@@ -125,6 +125,7 @@ export function findLeafNodeAtOffset(node: CstNode, offset: number): LeafCstNode
         let firstChild = 0;
         let lastChild = node.children.length - 1;
         while (firstChild <= lastChild) {
+            if (firstChild === lastChild) return findLeafNodeAtOffset(node.children[firstChild], offset);
```
Comment (spoenemann) [inline packages/langium/src/utils/cst-util.ts:128]: Suggestion: change the loop condition to `firstChild <= lastChild` and move this case to after the loop. That would separate the cases more clearly.

I'd also suggest to wrap the `if` body in a block with curly braces for the sake of consistency. I know we often used to write with a more compact style with Xtend, but I see consistency as more important than brevity nowadays.

Comment (insafuhrmann) [reply packages/langium/src/utils/cst-util.ts]: Thanks for the suggestion! I have implemented that (I assumend from context you meant "change the loop condition to `firstChild < lastChild`").

**Comment by insafuhrmann** [comment]
@spoenemann
Thanks for your suggestions, they are implemented now. I assumed that you meant that the loop condition should be changed to "less than", as otherwise there would be no suggested change and no separated cases, `<=` was the previous condition.

**Review by spoenemann — APPROVED** [review]
Oh yes, thanks!

**Pull request merged**